### PR TITLE
RHCOS4: Updated rule for rhel9 based RHCOS

### PR DESCRIPTION
**Pull request by Vincent056**
#### Description:

Added a new CPE for RHEL9 RHCOS, disabled `iptables` rule for rhel9-rhcos4, and added a new rule to detect if `iptables-nft` is being installed.

Fix rule `ensure_logrotate_activated`.

#### Rationale:

`iptables` is being replaced by `iptables-nft` in RHEL9 based RHCOS.

The newer logrotate use systemd timer instead of cron.

### CODE REVIEW

File: linux_os/guide/system/logging/log_rotation/ensure_logrotate_activated/oval/shared.xml [hunk 1]
```diff
@@ -8,8 +8,10 @@
       test_ref="test_logrotate_conf_daily_setting" />
       <criterion comment="check that there is no weekly/monthly/yearly keyword in logrotate.conf"
       test_ref="test_logrotate_conf_no_other_keyword" />
-      <criterion comment="Check if /etc/cron.daily/logrotate file exists (and calls logrotate)"
-      test_ref="test_cron_daily_logrotate_existence" />
+      <criteria comment="Check if either logrotate timer or cron job is eneabled" operator="OR">
+        <criterion comment="Check if /etc/cron.daily/logrotate file exists (and calls logrotate)" test_ref="test_cron_daily_logrotate_existence" />
+        <extend_definition comment="check kubeletconfig master" definition_ref="timer_logrotate_enabled" />
```
Comment (jhrozek) [inline linux_os/guide/system/logging/log_rotation/ensure_logrotate_activated/oval/shared.xml:13]: copy-paste?

File: linux_os/guide/system/logging/log_rotation/package_logrotate_installed/rule.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+documentation_complete: true
+
+title: 'Ensure logrotate is Installed'
```
Comment (jhrozek) [inline linux_os/guide/system/logging/log_rotation/package_logrotate_installed/rule.yml:3]: Where is this rule used?

Comment (Vincent056) [reply linux_os/guide/system/logging/log_rotation/package_logrotate_installed/rule.yml:3]: It's needed by timer enabled template

File: linux_os/guide/system/logging/log_rotation/ensure_logrotate_activated/oval/shared.xml (hunk 1 above)
Comment (Vincent056) [reply linux_os/guide/system/logging/log_rotation/ensure_logrotate_activated/oval/shared.xml:13]: lol, yes

**Comment by Vincent056** [comment]
> /test e2e-aws-rhcos4-high /test e2e-aws-rhcos4-e8 /test e2e-aws-rhcos4-moderate

I think I need to fix the CPE, probably have some issues with the regex matching, the other fix has worked.

**Comment by Vincent056** [comment]
Changed expected e2e test result.

**Review by jhrozek — APPROVED** [review]
LGTM

**Comment by jhrozek** [comment]
Actually, let's hold the merge, the automatus failures look related...

**Comment by jhrozek** [comment]
@Vincent056 it looks like we can ignore this error:
```
ERROR - Rule 'xccdf_org.ssgproject.content_rule_package_iptables-nft_installed' isn't present in benchmark 'xccdf_org.ssgproject.content_benchmark_RHEL-8' in '/tmp/ssgts-ds-ztez_y1t'
```
but what about this one?
```
 ERROR - Script logrotate_no_cron_daily.fail.sh using profile (all) found issue:
ERROR - Rule evaluation resulted in error, instead of expected fail during initial stage 
ERROR - The initial scan failed for rule 'xccdf_org.ssgproject.content_rule_ensure_logrotate_activated'.
```

**Comment by Vincent056** [comment]
> > @Vincent056 it looks like we can ignore this error:
> > ```
> > ERROR - Rule 'xccdf_org.ssgproject.content_rule_package_iptables-nft_installed' isn't present in benchmark 'xccdf_org.ssgproject.content_benchmark_RHEL-8' in '/tmp/ssgts-ds-ztez_y1t'
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > but what about this one?
> > ```
> >  ERROR - Script logrotate_no_cron_daily.fail.sh using profile (all) found issue:
> > ERROR - Rule evaluation resulted in error, instead of expected fail during initial stage 
> > ERROR - The initial scan failed for rule 'xccdf_org.ssgproject.content_rule_ensure_logrotate_activated'.
> > ```
> 
> systemd is not working in the fedora container, should we just ignore this issue, or remove this test? The e2e has passed, so the timer detection is working as expected.

```
System has not been booted with systemd as init system (PID 1). Can't operate.
Failed to connect to bus: Host is down
```

**Comment by jhrozek** [comment]
> systemd is not working in the fedora container, should we just remove this test? The e2e has passed, so the timer detection is working as expected.
I do not know why I missed this. Thank you.

**Pull request merged**